### ISSUE TITLE
[sdk] Update absl-py requirement to accommodate future TF versions

### ISSUE THREAD

**Issue by jiyongjung0** [kubeflow/pipelines#7171]
### Environment

*  KFP version:
N/A

*  KFP SDK version: 
1.8.10 (latest)

*  All dependencies version:
N/A

### Steps to reproduce

I believe that Tensorflow is frequently used with KFP and TF and should be able to be installed together. TF updated their absl-py requirement to `>=1.0.0` recently. [setup.py in TF](https://github.com/tensorflow/tensorflow/blob/c93e4757491a4abb7aa990e18f2d321e4a935a03/tensorflow/tools/pip_package/setup.py#L77). The next TF version(probably, 2.8.0) cannot be installed with the recent version kfp, because kfp is pinned to `<=0.11`. TF-nightly was already hit by this issue.

```
$ pip install tf-nightly==2.9.0.dev20220116 kfp==1.8.10
...
ERROR: Cannot install kfp==1.8.10 and tf-nightly==2.9.0.dev20220116 because these package versions have conflicting dependencies.

The conflict is caused by:
    tf-nightly 2.9.0.dev20220116 depends on absl-py>=1.0.0
    kfp 1.8.10 depends on absl-py<=0.11 and >=0.9

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
```

### Expected result

I think that the version range of absl should become `absl-py>=0.9,<2.0.0` if possible. Absl-py doesn't have [any breaking change recently ([Change log](https://github.com/abseil/abseil-py/blob/main/CHANGELOG.md)) and it should be safe to upgrade. We are doing [a similar upgrade in TFX](https://github.com/tensorflow/tfx/pull/4573).

It would be great if you can do a patch release with the fix if possible. Because we are already seeing failures in integration tests that utilizing TF-nightly and kfp.

### Materials and Reference
N/A

---

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by jiyongjung0** [comment]
@chensun Thank you for resolving the issue!

By the way, is there a plan to release a new version of kfp in the near future? TFX integration tests are still failing and I'm thinking about some kind of hacks to workaround the current breakage if the release will not happen shortly.

**Comment by chensun** [comment]
Yes, we plan to release 1.8.11 shortly, and I'll cherry pick your change into the release branch. Will let you know when the release is out.

**Comment by chensun** [comment]
@jiyongjung0 FYI, we released 1.8.11 today that contains the change you made: https://pypi.org/project/kfp/1.8.11/

**Comment by jiyongjung0** [comment]
I confirmed that our tests pass with the new version. Thank you for the release!
